docs/model_developers/modules.rst

```diff
@@ -122,3 +122,96 @@ The format for requires in requirements.txt is the folllowing:
+installed (either in editable or in packaged form). If you're working on a project, all required v1
+modules should be checked out in the ``modulepath`` directory.
+
+When you run a compile from the active Python environment context, the compiler will find both the
+v1 and v2 modules and use them for both their model and their plugins.
+
+Similarly, when you run a module's unit tests, the installed v2 modules will automatically be used
+by the compiler. As for v1 modules, by default, the ``pytest-inmanta`` extension makes sure the
+compile itself runs against an isolated project, downloading any v1 module dependencies at
+compile-time. If you want to compile against local versions of v1 modules, have a look at the
+``--use-module-in-place`` option in the ``pytest-inmanta`` documentation.
+
+
+Distributing modules
+====================
+This section is about v2 modules. V1 modules only require a version tag to be recognized as a
+released version. While a version tag is still good practice for v2 modules, it isn't sufficient
+to consider it released.
+
+You can package a v2 module with ``inmanta module build`` which will create an sdist and a bdist
+of the Python package. You can then publish this to the Python package repository of your choice,
+for example the public PyPi repository. For an inmanta project, follow the same procedure but
+substitute ``module`` with ``project``.
+
+The orchestrator server generally (see
+:ref:`Advanced concepts<modules-distribution-advanced-concepts`) installs both project and modules
+from the configured Python package repository, respecting the environment's version constraints on
+the project package, the project's constraints on its modules and all inter-module constraints. The
+server is then responsible for supplying the agents with the appropriate ``inmanta_plugins``
+modules.
+
+.. _modules-distribution-advanced-concepts
+
+Advanced concepts
+-----------------
+
+Freezing a project
+^^^^^^^^^^^^^^^^^^
+Prior to releasing a new stable version of an inmanta project, you might wish to freeze its module
+dependencies. This will ensure that the orchestrator server will always work with the exact
+versions specified. You can achieve this with
+``inmanta project freeze --recursive --operator "=="``. This command will freeze all module
+depdencies to their exact version as they currently exist in the Python environment. The recursive
+option makes sure all module depdencies are frozen, not just the direct dependencies. In other
+words, if the project depdends on module ``a`` which in turn depends on module ``b``, both modules
+will be pinned to their current version in ``pyproject.toml``.
+
+Manual export
+^^^^^^^^^^^^^
+The `inmanta export` command exports a project and all its modules to the orchestrator server.
+When this method is used, the orchestrator does not install any modules from the Python package
+repository but instead contains all code (both model and plugins) as present in the local Python
```

Comment:
inmanta export sends no model at all (only resources and python code) ?

Reply:
I think I misunderstood something then. Last week we discussed two modes of the server: pull from version control and manual export of a model. With the second mode I was referring to this, which is apparently not correct. What then is this second mode?
In any case, I'll update this to say it exports the Python code only.